src/compiler/parser.ts

```diff
@@ -2973,7 +2973,7 @@ namespace ts {
         function parseDelimitedList<T extends Node | undefined>(kind: ParsingContext, parseElement: () => T, considerSemicolonAsDelimiter?: boolean): NodeArray<NonNullable<T>> | undefined {

```

Comment:
IMO these two fixes should really have `T extends Node`, then refer to `T | undefined`, but, this is the more minimal fix.